kartothek/io_components/metapartition.py

```diff
@@ -1549,6 +1549,67 @@ def delete_from_store(self, dataset_uuid, store):
             store.delete(file_key)
         return self.copy(files={}, data={}, metadata={})
 
+    def get_parquet_metadata(self, store, table_name):
+        """
+        Retrieve the parquet metadata for the MetaPartition.
+        Especially relevant for calculating dataset statistics.
+
+        Parameters
+        ----------
+        store
+          A factory function providing a KeyValueStore
+        table_name
+          Name of the kartothek table for which the statistics should be retrieved
+
+        Returns
+        -------
+        pd.DataFrame
+          A DataFrame with relevant parquet metadata
+        """
+        if not isinstance(table_name, str):
+            raise TypeError("Expecting a string for parameter `table_name`.")
+
+        if callable(store):
+            store = store()
+
+        if self.files:
+            fd = store.open(self.files[table_name])
```

Comment:
File descriptors (even though they're likely virtual here) should always be guarded:

```suggestion
            with store.open(self.files[table_name]) as fd:
```